src/Random.cpp

```diff
@@ -69,11 +69,19 @@ std::uint32_t Random::randuint32() {
 }
 
 void Random::seedrandom(std::uint64_t seed) {
-    // Magic values from Pierre L'Ecuyer,
-    // "Tables of Linear Congruental Generators of different sizes and
-    // good lattice structure"
-    m_s[0] = (1181783497276652981ULL * seed);
-    m_s[1] = (1181783497276652981ULL * m_s[0]);
+
+    // Initialize state of xoroshiro128+ by transform the seed
+    // with splitmix64 algorithm.
+    // As suggested by http://xoroshiro.di.unimi.it/xoroshiro128plus.c
+
+    seed += 0x9e3779b97f4a7c15;
```

Comment:
Pull out the splitmix64 call rather than duplicating the code and constants twice.

Or just add a m_s[1] |= 1;